python/lsst/eotest/sensor/BFTask.py

```diff
@@ -0,0 +1,271 @@
+                ycorr_exp_err.append(np.mean(ycorr_amp_err))
+                mean_exp.append(np.mean(mean_amp))
+
+            #Store cov/var
+            xcorrs[amp] = xcorr_exp
+            ycorrs[amp] = ycorr_exp
+            means[amp] = mean_exp
+            BFResults[amp] = [xcorr_exp,xcorr_exp_err,ycorr_exp,ycorr_exp_err,mean_exp]
+
+        self.writeEotestOutput(BFResults, all_amps,sensor_id,meanidx)
+
+        return BFResults
+
+
+    def writeEotestOutput(self,BFResults,all_amps,sensor_id,meanidx=0):
+
+        outfile = os.path.join(self.config.output_dir,
+                               '%s_bf.fits' % sensor_id)
+        output = fits.HDUList()
+        output.append(fits.PrimaryHDU())
+        colnames = []
+        units = []
+        columns = []
+        for amp in all_amps:
+            colnames.extend(['AMP%02i_xcorr' % amp, 'AMP%02i_xcorr_err' % amp, 'AMP%02i_ycorr' % amp,
+            'AMP%02i_ycorr_err' % amp,'AMP%02i_MEAN' % amp])
+            units.extend(['Unitless', 'Unitless','Unitless','Unitless','ADU'])
+            columns.extend([np.array(BFResults[amp][0], dtype=np.float),
+                            np.array(BFResults[amp][1], dtype=np.float),
+                            np.array(BFResults[amp][2], dtype=np.float),
+                            np.array(BFResults[amp][3], dtype=np.float),
+                            np.array(BFResults[amp][4], dtype=np.float)])
+        formats = 'E'*len(colnames)
+        fits_cols = [fits.Column(name=colnames[i], format=formats[i],
+                                 unit=units[i], array=columns[i])
+                     for i in range(len(columns))]
+        output.append(fitsTableFactory(fits_cols))
+        output[-1].name = 'BF_STATS'
+        output[0].header['NAMPS'] = len(all_amps)
+        fitsWriteto(output, outfile, clobber=True)
+        #Output a file of the coefficients at a given mean, given
+        #as the index of the exposure in the list.
+
+        results_file = os.path.join(self.config.output_dir,
+                                        '%s_eotest_results.fits' % sensor_id)
+
+
+        results = EOTestResults(results_file, namps=len(all_amps))
+
+        for amp in all_amps:
+            results.add_seg_result(amp, 'xcorr', BFResults[amp][0][meanidx])
+            results.add_seg_result(amp, 'xcorr_err', BFResults[amp][1][meanidx])
+            results.add_seg_result(amp, 'ycorr', BFResults[amp][2][meanidx])
+            results.add_seg_result(amp, 'ycorr_err', BFResults[amp][3][meanidx])
+            results.add_seg_result(amp, 'MEAN', BFResults[amp][4][meanidx])
+
+
+        results.write(clobber=True)
+
+    def prep_image(self,exp,dark_image):
+        """
+        Crop the image to avoid edge effects based on the Config border
+        parameter. Additionally, if there is a dark image, subtract.
+        """
+
+        # Clone so that we don't modify the input image.
+        local_exp = exp.clone()
+        del exp
```

Comment:
I don't think `exp` needs to be deleted.  In any case, any deletion should be handled by the code that calls this function, otherwise clients using this code may be surprised that their object was deleted.